### DISCUSSION TITLE
Get filetype of REPL

### DISCUSSION THREAD

**Discussion by rwijtvliet** [Vigemus/iron.nvim#377] [Q&A]
I'd like to write keymaps that work both in the buffer with the source code and in the buffer with the REPL terminal. 

For example, `<leader>rf` should focus the REPL - regardless of where the cursor is currently located. (a) If it's in the code buffer, it should (if needed) make the REPL visible and move the cursor there, and (b) if it's already in the REPL buffer, nothing should happen.

These do not work in situation (b) (`Executing Lua callback: ...t/.local/share/nvim/lazy/iron.nvim/lua/iron/lowlevel.lua:39: Empty filetype`):

```lua
      vim.keymap.set('n', '<leader>rf', '<cmd>IronFocus<cr>')
      vim.keymap.set('n', '<leader>rf', function() iron.focus_on(vim.bo.filetype) end)
```

Same when trying to hide the REPL window with `IronHide` or `iron.hide_repl`.

--> ***Is there a way, a lua function, that I can use to find the filetype of the REPL buffer?***

**Comment by frere-jacques** [comment]
I stumbled over the same topic, because I was confused about hide repl giving errors if on the repl.

I think one can enhance this in the plugin itself and maybe will create a pull request.

If you want to do this for own mappings, you can require iron.lowlevel and get the store. There you have the configs for the different repls. You can iterate over them and check if the buffer matches your current buffer. If so you can grap the key of that config and have the filetype associated with that repl.

Thst is written down in iron.lowlevel.get_repl_ft_for_bufnr

**Comment by noprompt** [comment]
You can set in the `repl_open_cmd` function. I recently did this so I can [integrate iron with edgy](https://github.com/folke/edgy.nvim/discussions/112). Here's the relevant iron configuration from the referenced discussion:

```lua
{
  config = {
    repl_open_cmd = function(bufnr)
        -- iron doesn't set the filetype for the buffers it creates so we need
        -- to provide one.
        vim.api.nvim_set_option_value("filetype", "iron", { buf = bufnr })
        -- Create a window and return it's id.
        return require("iron.view").split.vertical.botright(40)(bufnr)
      end,
    }
}
```

@rwijtvliet  You could combine this with function @frere-jacques  suggested and create the filetype based on `iron.lowlevel.get_repl_ft_for_bufnr`, e.g. `iron-python`, `iron-sh`, etc.

**Reply by rwijtvliet** [reply]
Ah, awesome, many thanks! I'll have to postpone trying it - corporate has locked-down my laptop so much it's a real struggle getting neovim to run on it :/

**Reply by noprompt** [reply]
Ugh! That's a bummer. Well, hopefully that'll get sorted out and you'll have this to come back to. 😄
